frameworks/spark/src/main/java/com/criteo/hadoop/garmadon/spark/listener/GarmadonSparkListener.java

```diff
@@ -77,6 +78,43 @@ private void sendExecutorStateEvent(long time, State state, String executorId, S
         this.eventHandler.accept(time, buildOverrideHeader(executorId), executorStateEvent.build());
     }
 
+    private void sendApplicationStateEvent(long time, State state) {
```

Comment:
I would give a try using the constants in core/src/main/scala/org/apache/spark/internal/config/package.scala to access the conf values because they can provide the default values as well

Reply:
Those constants are private, I cannot reference them from the listener